### ISSUE TITLE
Support native BigInt for Node.JS >10.4

### ISSUE THREAD

**Issue by simlevesque** [awslabs/amazon-qldb-driver-nodejs#91]
Hi ! We'd like to use native BigInts with QLDB in the backend, any plans for that ?

**Comment by alpian** [comment]
Hi @simlevesque, 

Thanks for reaching out to us!

Are you referring to having a mechanism to automatically convert from a [NodeJS BigInt](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt) to an Ion `int`? We don't have this work "planned" right now but we could certainly look to add it. Could you possibly provide an example of what you would like the driver to do?

Thanks!
-Ian.

**Comment by omichowdhury** [comment]
I'd like to +1 this

Use case is pretty simple - serialisation and deserialisation from ion should be able to use BigInts instead of Numbers. 

I think this PR adds support to `ion-js` so hopefully this is mostly a dependency update.
https://github.com/amzn/ion-js/pull/721

**Comment by amzn-paunort** [comment]
Hi @omichowdhury, thanks for your interest in this. The PR you linked was merged into the in-development 5.0.0 branch of ion-js. Perhaps we can revisit it once it is released to the public.

As they noted in their PR discussion as well, moving to native `bigint` is a breaking change for customers, compared to `number`. This would have been nice to include in release 3.0.0 as part of a major version release, but without the ion-js support we could not include it. We may require another major version release to support this.

If the ion-js release allows us to simply upgrade the dependency and receive this, it would be easy enough for us to add. However, I can't say now either way whether it's something we will support when that opportunity arrives, nor can I give any sort of timeline on that.

**Comment by omichowdhury** [comment]
A major version is completely fine since adding support for BigInt is far less work than switching to JSBI throughout a codebase. 

Would you be able to poke the `ion-js` team to see what the 5.0.0 release is waiting on? It's been about 6 weeks since that PR was merged.

I also noticed `ion-hash-js` also has a dependency on `ion-js` so not sure if that would also be a drop in change or not. Looks like some minor changes would be needed, but it seems to mostly delegate int handling down to `ion-js`
